qiita_pet/handlers/analysis_handlers.py

```diff
@@ -215,8 +217,27 @@ def get(self):
         analyses = [Analysis(a) for a in
                     user.shared_analyses | user.private_analyses]
 
+        # Check if the request came from a local source
+        is_local_request = is_localhost(self.request.headers['host'])
+        mappings = {}
+        bioms = {}
+        for analysis in analyses:
+            _id = analysis.id
+            mapping = analysis.mapping_file
+            if mapping is not None:
+                mappings[_id] = download_link_or_path(
+                    is_local_request, mapping,
+                    get_filepath_id('analysis', mapping), 'mapping file')
+            else:
+                mappings[_id] = ''
+            bioms[_id] = '\n'.join(
```

Comment:
sure, but that spans lines and multiple line comprehensions are semantically complex